### PR TITLE
add graph to OTF ADR and tox env for updating plantuml

### PR DESCRIPTION
**Pull request by DropD**
## Description

This PR adds the workflow reference graph from the recent internal presentation to the OTF toolchain ADR, exploring a possible standard way to handle graphs in ADRs (based on https://lukemerrett.com/c4-diagrams-as-code-architectural-joy/).

In addition to embedding `plantuml` inside the markdown and linking to the generated SVG file, this PR adds tox automation for updating all ADR plantuml graphs in a standardized way. This currently relies on plantuml CLI being pre-installed and is merely one possible way. Alternatives:

- https://github.com/danielyaa5/puml-for-markdown: does not allow to embed puml directly inside markdown but creates and embeds / linksJK graphs automatically from links to puml files. It does this by using the plantuml server to create a link to the rendered graph. No storing svg files in the repo.
- multiple github actions are available on the Marketplace: https://github.com/marketplace?query=plantuml
- potentially other ones listed at https://plantuml.com/running

### CODE REVIEW

**Comment by egparedes** [comment]
bors try

**Review by egparedes — CHANGES_REQUESTED** [review]
It is almost ready to go. There are just a couple of typos and the renaming of the tox action.

File: tox.ini [hunk 1]
```diff
@@ -51,3 +51,21 @@ commands =
     jupytext QuickstartGuide.md --to .py
     python QuickstartGuide.py
 commands_post =
+
+[testenv:dev-adr-plantuml]
```
[the remark this answers is not in the document]
Comment (DropD) [reply tox.ini]: Noted, thanks!

**Review by egparedes — APPROVED** [review]
It looks good to me.

**Comment by egparedes** [comment]
Why are CI tests not running in this PR anymore?

**Comment by havogt** [comment]
No idea, seems to work after another merge with functional.

**Pull request merged**